homeassistant/components/sensor/dominionenergy.py

```diff
@@ -0,0 +1,49 @@
+"""Support for Washington Dominion Energy."""
+
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setting the platform in HASS and getting the username and password."""
+    add_devices([DominionEnergySensor(config['username'], config['password'])])
+
+
+class DominionEnergySensor(Entity):
+    """Washington Dominion Energy Sensor will check the bill on daily basis."""
+
+    from datetime import timedelta
+    HOURS_TO_UPDATE = timedelta(hours=24)
+    CURRENT_BILL_SELECTOR = str("#homepageContent > div:nth-child(3) >" +
+                                " div:nth-child(2) > p > span")
+
+    def __init__(self, username, password):
+        """Initialize the sensor."""
+        self._state = None
+        self._password = password
+        self._username = username
+
+    @property
+    def name(self):
+        """Name of the Sensor: DominionEnergy Sensor."""
+        return "Dominion Energy"
```

Comment:
The name should be a configuration option (`CONF_NAME`) to allow users to set their own name. "Dominion Energy" can be the default for the optional configuration option.

Reply:
do i have to set default value if self._name is empty or this is handled by hass by taking the file name ? @fabaff